File: react-native-core-client/ios/CoreClient.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(CoreClient, NSObject)

RCT_EXTERN_METHOD(hi:
                 (RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(fetchLaunchRockets:(BOOL)force
                 withResolver:(RCTPromiseResolveBlock)resolve
                 withRejecter:(RCTPromiseRejectBlock)reject)

+ (BOOL)requiresMainQueueSetup
{
  return NO;
}

@end

File: react-native-core-client/ios/CoreClient.swift
import shared

@objc(CoreClient)
class CoreClient: NSObject {
    let sdk = CoreClientManager.Builder(databaseDriverFactory: DatabaseDriverFactory()).build()

    @objc(hi:withRejecter:)
    func hi(_ resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) -> Void {
      resolve("hiii")
    }

    @objc(fetchLaunchRockets:withResolver:withRejecter:)
    func fetchLaunchRockets(force: Bool, resolve: @escaping RCTPromiseResolveBlock, reject:@escaping RCTPromiseRejectBlock) -> Void {
        sdk.fetchLaunchRockets(forceReload: true)
                        .onSuccess { result in
                            guard let rocketsResult = result else {return}
                            resolve(RocketLaunch.companion.toJson(array: rocketsResult))
                        }.onFailure { KotlinThrowable in
                            reject("empty", "empty", KotlinThrowable.asError())
                        }
    }
}
